Source/CombatExtended/CombatExtended/Gizmos/Command_ArtilleryTarget.cs

```diff
@@ -169,6 +175,15 @@ public override void ProcessInput(Event ev)
                     }
                 }
                 return distanceMessage + "\n" + "CE_ArtilleryTarget_ClickToOrderAttack".Translate() + extra;
+            }, canSelectTarget: (targetInfo) =>
+            {
+                if (!targetInfo.HasWorldObject || targetInfo.Tile == turretTile ||
+                        !(targetInfo.WorldObject.def == WorldObjectDefOf.Ambush) &&
```

Comment:
```suggestion
                        (targetInfo.WorldObject as MapParent)?.Map == null
```
Why only ambush? Think it's better

Everything else is good to go